### PR TITLE
Refs #1977 - Don't rely on Puppetversion to determine report version

### PR DESCRIPTION
**Pull request by GregSutcliffe**

### CODE REVIEW

File: app/models/report.rb [hunk 1]
```diff
@@ -82,16 +82,13 @@ def self.import(yaml)
       # parse report metrics
       raise "Invalid report: can't find metrics information for #{host} at #{report.id}" if report.metrics.nil?
 
-      test_env    = Puppet::PUPPETVERSION.to_i >= 3 ? :@environment       : '@environment'
-      test_format = Puppet::PUPPETVERSION.to_i >= 3 ? :@report_format     : '@report_format'
-      test_status = Puppet::PUPPETVERSION.to_i >= 3 ? :@resource_statuses : '@resource_statuses'
 
       case
-      when report.instance_variables.include?(test_env)
+      when report.instance_variables.any? {|v| v.to_s == "@environment"}
```
Comment (ohadlevy) [inline app/models/report.rb]: you can use detect too (afair it stops once it finds match vs looping the entire thing)

**Comment by domcleal** [comment]
+1, should be much more robust.

**Comment by GregSutcliffe** [comment]
@ohadlevy Cool will fix that later today

**Comment by GregSutcliffe** [comment]
Done. I've also included a small change @witlessbird made in his 193 branch - seems sensible to me.

**Comment by ohadlevy** [comment]
does this closes #1977 or refs it?

**Comment by ohadlevy** [comment]
merged as fix for #1997 - thanks!

**Pull request closed without merging**